internal/manifests/collector/ingress.go

```diff
@@ -45,24 +45,15 @@ func Ingress(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemet
 		return nil
 	}
 
-	pathType := networkingv1.PathTypePrefix
-	paths := make([]networkingv1.HTTPIngressPath, len(ports))
-	for i, p := range ports {
-		paths[i] = networkingv1.HTTPIngressPath{
-			Path:     "/" + p.Name,
-			PathType: &pathType,
-			Backend: networkingv1.IngressBackend{
-				Service: &networkingv1.IngressServiceBackend{
-					Name: naming.Service(otelcol.Name),
-					Port: networkingv1.ServiceBackendPort{
-						// Valid names must be non-empty and no more than 15 characters long.
-						Name: naming.Truncate(p.Name, 15),
-					},
-				},
-			},
-		}
+	var rules []networkingv1.IngressRule
+	switch otelcol.Spec.Ingress.RuleType {
+	case v1alpha1.IngressRuleTypePath, "":
+		rules = []networkingv1.IngressRule{createPathIngressRules(otelcol.Name, otelcol.Spec.Ingress.Hostname, ports)}
+	case v1alpha1.IngressRuleTypeSubdomain:
+		rules = createSubdomainIngressRules(otelcol.Name, otelcol.Spec.Ingress.Hostname, ports)
 	}
 
+	createSubdomainIngressRules(otelcol.Name, otelcol.Spec.Ingress.Hostname, ports)
```

Comment:
this is a no-op right now, is that intentional?

Reply:
removed